.github/actions/nuxeo/nos-publish/action.yml

```diff
@@ -0,0 +1,128 @@
+# A github composite action to publish package to Nuxeo Online service using a form post request.
+name: 📦 Publish to Nuxeo Online Services
+description: |
+  Publish package to Nuxeo Online service.
+inputs:
+  package-path:
+    description: Path to the package file to be published.
+    required: true
+  nos-username:
+    description: Nuxeo Online Service username for authentication.
+    required: true
+  nos-token:
+    description: Nuxeo Online Service token for authentication.
+    required: true
+  nos-env:
+    description: |
+      Nuxeo Online Service target environment (e.g., production, staging).
+      If not set, it will be determined based on the branch name.
+  nos-org:
+    description: |
+      Nuxeo Online Service organization name.
+    default: nuxeo
+  skip-verify:
+    description: Skip verification of the published package.
+    default: "false"
+outputs:
+  publishing-status:
+    description: Status of the publishing process.
+    value: ${{ steps.publish-output.outputs.status }}
+  package-url:
+    description: URL of the published package on Nuxeo Online Services.
+    value: ${{ steps.publish-output.outputs.url }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Extract package info
+      shell: bash
+      env:
+        PACKAGE_XML: "package.xml"
+      run: |
+        unzip -t "${{ inputs.package-path }}" $PACKAGE_XML || {
+          echo "❌ The package file '${{ inputs.package-path }}' has no valid descriptor"
+          exit 1
+        }
+        echo "📦 Extracting package information from ${{ inputs.package-path }}..."
+        for i in name version; do
+          VALUE=$(unzip -p "${{ inputs.package-path }}" $PACKAGE_XML | yq -p=xml ".package.+@${i}")
+          export "PACKAGE_${i^^}=${VALUE}"
+          echo "PACKAGE_${i^^}=${VALUE}" | tee -a $GITHUB_ENV
+        done
+        if [[ -z "${PACKAGE_NAME}" || -z "${PACKAGE_VERSION}" ]]; then
+          echo "❌ Unable to extract package name or version from descriptor"
+          exit 1
+        fi
+
+    - name: Get Branch name
+      uses: Alfresco/alfresco-build-tools/.github/actions/get-branch-name@v10.1.0
+
+    - name: Evaluate Target env if none has been set
+      shell: bash
+      run: |
+        case "${{ inputs.nos-env }}" in
+          "production")
+            CONNECT_HOST=connect.nuxeocloud.com
+            ;;
+          "staging")
+            CONNECT_HOST=nos-preprod-connect.nuxeocloud.com
+            ;;
+          *)
+            if [[ "${BRANCH_NAME}" == "${{ github.event.repository.default_branch }}" ]]; then
+              CONNECT_HOST=connect.nuxeocloud.com
+            else
+              CONNECT_HOST=nos-preprod-connect.nuxeocloud.com
+            fi
```

Comment:
Could this be an issue if the `nos-env` input is undefined and BRANCH_NAME is a reference branch which is not the default one, e.g. [lts-2025](https://github.com/nuxeo/nuxeo-3d-viewer/tree/lts-2025) in nuxeo-3d-viewer? In this case we would want to upload to Connect prod.